### ISSUE TITLE
HTTPS agent: stop using entire ca / pfx / cert object in socket names

### ISSUE THREAD

**Issue by othiym23** [nodejs/node-v0.x-archive#7264]
For my application, I'm using an array of 80 CA certificates (ASCII-encoded PEM format) to initialize the HTTPS agent. The current implementation of `.getName()` just appends the whole property, which means that the names of my sockets are HUGE.

This can't be efficient, either in terms of memory or speed. I'd suggest using a fast hash, but but getName() gets called every time a request is added or a connection is freed, so I don't know if that would be fast enough. @trevnorris, any ideas?

**Comment by jasnell** [comment]
@othiym23 ... any further thoughts on this?

**Comment by othiym23** [comment]
Yes. It would be great to get this fixed. ;) I've since moved on from New Relic, but I don't think this is any less of a performance footgun than it was when I first noticed it. I'm also not sure whether this has been addressed in the interim, and it's possible that some of the other changes to how HTTPS agents are managed has addressed the underlying problem.

I'm also not sure this counts as a feature request, as the bug is pretty obvious if you're using more than even a couple custom CA certs (which doesn't strike me as an unreasonable requirement in many corporate environments with their own PKI / TLS infrastructure).
